### PR TITLE
Extend regression flow with trial and course connections

### PR DESCRIPTION
**Pull request by jstvz**
This PR adds `deploy_trial_config` and `enable_course_connections` tasks to `config_regression`.

----

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

# Testing Notes

``` console
$ cci flow run regression_org --org release
```

### CODE REVIEW

File: cumulusci.yml [hunk 1]
```diff
@@ -99,6 +99,17 @@ flows:
...
                 task: deploy_k12_kit_settings
 
+    config_regression:
+        steps:
+            2:
+                task: enable_course_connections
+                options:
+                    managed: True
+            3:
+                task: deploy_trial_config
```
Comment (davisagli) [inline cumulusci.yml]: @jstvz does this deploy anything that the admin profile needs permissions for? if so you'll need update_admin_profile afterward.

Comment (jstvz) [reply cumulusci.yml]: No, it's only deploying the admin profile itself. That said I re-ordered the steps to future-proof it.

**Comment by davisagli** [comment]
@jstvz what's the reasoning for adding this to config_regression but not config_managed?

File: cumulusci.yml (hunk 1 above)
Comment (jofsky) [reply cumulusci.yml]: @jstvz are there defaults for unmanaged and managed values if they're not set?

**Comment by jstvz** [comment]
Primarily because I want to keep the `install_*` flows closer to the installer's output, with `trial_org` providing more configuration as necessary. I'm willing to entertain counter-arguments, though.

File: cumulusci.yml (hunk 1 above)
Comment (jstvz) [reply cumulusci.yml]: @jofsky: Yes, you can see them via `cci task info`.

**Comment by davisagli** [comment]
@jstvz ok. That's a different choice from what we did for NPSP, so maybe worth discussing. In retrospect install_prod should have been named prod_org and install_prod_no_config should have been called install_prod, but I haven't quite seen the path to cleaning it up until we make a backwards-incompatible release.

**Pull request merged**